### DISCUSSION TITLE
NSPanel Customisation

### DISCUSSION THREAD

**Discussion by vitalikkh** [Blackymas/NSPanel_HA_Blueprint#1502] [Q&A]
First of all - Happy New Year! and many BIG Thanks to @Blackymas for helping me to setup and configure NSPanel, US version.
So, Panel is up and running, all components are up to dated (everything is on Version 4.1.1) but I am very confused how to configure
this toy to my very specific needs. On a first glance it looked like Blueprint GUI should/could do everything what I am looking for.
However after playing a bit with Blueprint configuration I realized it is not that easy and intuitive as I was expected.
First let me explain what is my target application.
The Project "as is" is loaded with many very nice and I am sure, very useful features but my immediate needs is limited for
creating very simple Wall Mounted Scene Controller with 6-8 user customizable Buttons (I am looking for ability to add custom
Icon and Label for each Button). And, YES this is doable! But here is what is not very intuitive (but I may not be doing it right):
- It is not intuitive how the Generic Blueprint is assigned to the Specific NSPanel. It took me a while to figure this out but
  now I am all set know the trick.
- Labels for the Hardware Buttons are only visible on a Home Page. Why? These labels absolutely must be visible on EVERY
   Page because they are a Description for the Physical Hardware Buttons which are ALWAYS present and functional.
- The same comment is for the Relay Status. These are also Always Present Physical Devices and their Status (and Labels
   which is currently not a case) should/must be ALWAYS visible on any page (not only Home Page).
   Some Icons are listed but cannot be displayed (may be missing in a library. For instance "mdi:faucet").
- I selected startup page to be a buttonpage02:
   ![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/105172911/ef236c58-d882-4a34-9b91-e961dff86c3b)
   but on the Power Cycle (i.e. reboot) panel always starts with Home Page.

OVERAL,
I am very happy with this project and I am very glad I found it. But it will be nice and helpful to have very simple ether Blueprint
or just an ability to configure a panel with a simple YAML (or so). This already could be case but I am not a software person and
simply could be missing a point. It will be very nice if someone could point me in a right direction.
Sure, I may have more questions in a future.

Bottom line:
I like this project "as is".

Thank you very much and Happy New Year!,

- Vitaliy

UPDATE 1
The ability to change Icon Color on a Button is very nice feature but because when Button Status is
ON the background color is Bright White (and could not be changed?) the Icons are barely visible
unless Icon Color is Black. This makes an ability to customize Icon Color somewhat useless.
 
UPDATE 2
@Blackymas 

1. Is it possible to change a Background Color when Button is in ON state?
    Current Background Color is very bright white. This basically make an Icon alsmost
    invisible unless Icon color is Black.
3. Is it possible to force a displayed page by Command?
    Entity "nspanel Current page" shows what is a current page but I am looking for
    the ability to set it by command.

**Comment by edwardtfn** [comment]
- #1516 
- #1517 
- #1518 
- #1519 
- #1520 
- #1521 
- #1522

**Comment by edwardtfn** [comment]
> I am very happy with this project and I am very glad I found it. But it will be nice and helpful to have very simple ether Blueprint
or just an ability to configure a panel with a simple YAML (or so). This already could be case but I am not a software person and
simply could be missing a point. It will be very nice if someone could point me in a right direction.
Sure, I may have more questions in a future.

Could you please elaborate more on this?
Was difficult to setup the yaml in your case?
What was the difficult part?

**Reply by vitalikkh** [reply]
Home Assistant is not my primary Home Automation Controller/Hub (primary Home Automation hubs are Hubitat Elevation C8+C7).
My primary use for the Home Assistant is to add a support for a Devices which are not directly supported by Hubitat Elevation (HE).
NSPanel is one of these devices. Yours project is EXCELENT! It will solve a problem how to create very flexible Multi-button wall
mounted Scene Controller with ability for labeling all buttons electronically. For my specific needs the project "as is' is over killer.
But "YES", I am learning how to use it (I guess, very successfully) and making very good progress. First panel is already almost
configured and what is very important in my case - it is very well integrates with HE.
At the beginning few things was not very intuitive but once I figured out what is what everything became more or less clear.
First big confusion was to see a RENAME prompt after I made some changes in Blueprint and clicked on SAVE button. It took me
a while to realize that this way a device-specific automation YAML file was created and next time I have to use this created YAML
file to add or modify something in the device configuration. I guess, instead of RENAME more intuitive and self-expenatory will
be something like "Create a device-spesific automation YAML file". It could be too long but certainly will be far more intuitive.
For the improvements I would like to see next changes (few I already mentioned in my original post):
- Selectable background for the Home Page and ideally for all pages. Frankly my preference is just a plain solid color (no pictures);
- Labels and Status for the HW Buttons and Relays should/must be always visible (now they are visible only on a Home Page);
  Right now I am not using Chips but my good guess, Chips also better to be always visible;
- Button Background Color for the ON condition is very bright white. This way only good visible color for Icons is Black.
  Icon color is user selectable which is very good  but unfortunately next to useless because all colors other than Black is near invisible.
  It will be nice to have user selectable background color for the OFF condition  as well.

But as I already mentioned, I am very happy with the project "as is".

PS.
I got many emails notifications regarding my post and frankly I am lost in this forest of messages.
Question.
For the project update do I have to reflash the entire project?

**Reply by edwardtfn** [reply]
> First big confusion was to see a RENAME prompt after I made some changes in Blueprint and clicked on SAVE button. It took me
a while to realize that this way a device-specific automation YAML file was created and next time I have to use this created YAML
file to add or modify something in the device configuration. I guess, instead of RENAME more intuitive and self-expenatory will
be something like "Create a device-spesific automation YAML file". It could be too long but certainly will be far more intuitive.

I have no idea what is this **RENAME** you are talking about. 😆 
Could you please share a screen shot?

**Reply by vitalikkh** [reply]
On the very first attempt to use the default Blueprint after clicking on the blue SAVE button at the right bottom of the screen:
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/105172911/4004168f-5caf-491a-bf20-f9efa848e749)

**Reply by edwardtfn** [reply]
Aha!
Unfortunately this is part of Home Assistant and it is out of our control/influence. This is a standard dialog from Home Assistant.
And I agree, the **Rename** is not very intuitive here. It would be better **Save** in this case.
